packages/nodes-base/nodes/CalendlyV2/CalendlyTrigger.node.ts

```diff
@@ -0,0 +1,209 @@
+
+export class CalendlyTrigger implements INodeType {
+	description: INodeTypeDescription = {
+		displayName: 'Calendly Trigger',
+		name: 'calendlyV2Trigger',
+		icon: 'file:calendly.svg',
+		group: ['trigger'],
+		version: 1,
+		description: 'Starts the workflow when Calendly events occur.',
+		defaults: {
+			name: 'Calendly Trigger',
+		},
+		inputs: [],
+		outputs: ['main'],
+		credentials: [
+			{
+				name: 'calendlyV2Api',
+				required: true,
+			},
+		],
+		webhooks: [
+			{
+				name: 'default',
+				httpMethod: 'POST',
+				responseMode: 'onReceived',
+				path: 'webhook',
+			},
+		],
+		properties: [
+			{
+				displayName: 'Events',
+				name: 'events',
+				type: 'multiOptions',
+				options: [
+					{
+						name: 'invitee.created',
+						value: 'invitee.created',
+						description: 'Receive notifications when a new Calendly event is created',
+					},
+					{
+						name: 'invitee.canceled',
+						value: 'invitee.canceled',
+						description: 'Receive notifications when a Calendly event is canceled',
+					},
+				],
+				default: [],
+				required: true,
+			},
+			{
+				displayName: 'Scope Name or ID',
+				name: 'scope',
+				type: 'options',
+				default: '',
+				typeOptions: {
+					loadOptionsMethod: 'getScopeOptions',
+				},
+				required: true,
+			},
+		],
+
+	};
+
+	methods = {
+		loadOptions: {
+			async getScopeOptions(this: ILoadOptionsFunctions): Promise<INodePropertyOptions[]> {
+				const organizationScopeRoles = ['owner', 'admin'];
+				const resultData: INodePropertyOptions[] = [{
+					name: 'User',
+					value: 'user',
+					description: 'Receive notifications for your scheduled / canceled events',
+				}];
+				const { resource } = await calendlyApiRequest.call(this, 'GET', '/users/me', {});
+
+				const { collection: [membership] } = await calendlyApiRequest.call(this, 'GET', '/organization_memberships', {}, {
+					user: resource.uri,
+				});
+
+				if (organizationScopeRoles.includes(membership.role)) {
+					resultData.push({
+						name: 'Organization',
+						value: 'organization',
+						description: 'Receive notifications for all scheduled / canceled events in your organization',
+					});
+				}
+
+				return resultData;
+			},
+		},
+	};
+
+	// @ts-ignore (because of request)
+	webhookMethods = {
+		default: {
+			async checkExists(this: IHookFunctions): Promise<boolean> {
+				const webhookUrl = this.getNodeWebhookUrl('default');
+				const webhookData = this.getWorkflowStaticData('node');
+				const events = this.getNodeParameter('events') as string;
+				const scope = this.getNodeParameter('scope') as string;
+
+				const { resource } = await calendlyApiRequest.call(this, 'GET', '/users/me', {});
+
+				// Check all the webhooks which exist already if it is identical to the
+				// one that is supposed to get created.
+				const endpoint = '/webhook_subscriptions';
+				const query = {
+					scope,
+					organization: resource.current_organization,
+					count: 100,
```

Comment:
I'm not sure what the best practice is here for determining if a webhook already exists. The current [webhook subscriptions api ](https://developer.calendly.com/api-docs/faac832d7c57d-list-webhook-subscriptions)does not allow you to search by url or the webhook state. 

There could be many webhooks for the Calendly organization that are disabled so I didn't want to send too many requests here. For now I just check the first 100 webhook subscriptions that are returned.